utils/retrieve-state/src/lib.rs

```diff
@@ -314,8 +387,37 @@ pub mod offline {
         block_files
     }
 
+    /// Look for a block file matching the given [`BlockIdentifier`], returns `None` if not found.
+    pub fn get_block_file(
+        chain_path: impl AsRef<Path>,
+        block_identifier: &BlockIdentifier,
+    ) -> Option<DirEntry> {
+        walkdir::WalkDir::new(chain_path)
+            .into_iter()
+            .filter_map(|entry| entry.ok())
+            .find(|entry| {
+                let file_name = match entry.file_name().to_str() {
+                    Some(file_name) => file_name,
+                    None => return false,
+                };
+                let split = file_name.split('-').collect::<Vec<&str>>();
+                if let ["block", height, hash] = &split[..] {
```

Comment:
Detection of the block's height and hash is duplicated across a couple methods. It would be a more declarative approach if there was a method that took a file name and returned an object wrapping the file but that exposes the underlying block's hash and height. It could be reused in all of the places where we manually split the file name and pattern match on the parts.

Reply:
I don't disagree with this perspective or with other comments in the vein of factoring this stuff out, but I think I'd like to do that work as a part of #2121.

Reply:
#2121 as well